### PR TITLE
Group8/voting methods

### PR DESCRIPTION
**Pull request by charlie-ccw**
Here are the six voting methods, this file just provides the functions of each method instead of affecting the game loop

### CODE REVIEW

File: internal/common/voting/VotingMethods.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+	for key, value := range voteCount {
+		if value > maxScore {
+			winner = key
+			maxScore = value
+		}
+	}
+
+	return winner
+}
+
```
Comment (WillPowellUk) [inline internal/common/voting/VotingMethods.go:175]: Same with this method - doesn't this require the agents to recursively revote? As in the vote is repeated again and again with one fewer agents to vote for each time.

File: internal/common/voting/VotingMethods.go [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+			maxVotes = votes
+			winner = lootBox
+		}
+	}
+
+	// return the final winner
+	return winner
+}
+
+func Runoff(voteList []map[uuid.UUID]float64) uuid.UUID {
```
Comment (WillPowellUk) [inline internal/common/voting/VotingMethods.go:49]: I could well have misinterpretted this voting method, but doesn't this require the agents to revote for round two?

**Review by WillPowellUk — CHANGES_REQUESTED** [review]
Nice work! Code looks v. clean. Could you just confirm that run off and instant run off doesn't require the agents to revote? 

Would you be able to make a switch case with enums (iota) that can be used to switch between these different voting methods in the server? So for experimentation we can simply toggle on which voting mechanism to use?

File: internal/common/voting/VotingMethods.go (hunk 1 above)
Comment (charlie-ccw) [reply internal/common/voting/VotingMethods.go:177]: Revote is not needed for every method

**Pull request closed without merging**